### DISCUSSION TITLE
Build call with FirewallException fails on v5

### DISCUSSION THREAD

**Discussion by jothalha** [wixtoolset/issues#8646] [Questions]
I've recently looked into running wix through wine on linux and simultaniously upgraded from Wix v3 to v5. My basic structure that used to work in v3 is the following

```xml
<File Id="<EXE_FILE>" Source="<PATH>" KeyPath="yes">
  <fire:FirewallException Id="MyFirewallException" Name="Foo" Description="Bar" Protocol="udp" Scope="any" />
</File>
```

wix.exe returns `error WIX0136: There was an error importing the file ...\_idts\Wix5FirewallException.idt` but no additional output.

~~One thing to note is that with v4 the rule seems to also work as expected.~~ With v4 no errors are emitted, but the firewall rule fails to install. Not sure if this is a usage error on my path, but based on no similar reports I assume so. I tried different attributes to see if it would work at some point. The appended idt file (as text) is not exactly the same as the above definition (i tried to add action and some others), but still failed with the same error. If someone could point me in the right direction, I would really appreciate it.

[Wix5FirewallException.txt](https://github.com/user-attachments/files/16324977/Wix5FirewallException.txt)

**Comment by jothalha** [comment]
Had a Win system now to test this on, seems like a linux/wine specific issue. I guess a workaround would be a custom action. Not sure if anyone has an idea why building fails via wine.

**Reply by bevanweiss** [reply]
Does Wine use the Windows DotNet, or is it using Mono?  It's possible there's some handling differences between the platforms.
You're probably better off running a VM with Windows in it for these situations.
There's no test cases that are being exercised under any Wine environments, everything is tested under pure Windows, so that's the safe space to stay in.
I'm sure if you wanted to build up such additional test environments (i.e. some CI pipeline as a Github action) which did the Wine testing (and was successfully passing those tests) it would be appreciated.  But it does sound like a lot of work just to avoid running pure Windows.

**Reply by jothalha** [reply]
i installed dotnet. there may definitely be handling differences, but also the underlying wine msi functions will not be the same I think.

i may look into it a bit further if I have time, but with the recent experience I tend to agree with you here that a Win VM is the easier way. I'll stay with v3 for now - validation for light still has to be disabled there but at least it works. Thank you for responding

**Comment by marcobudde** [comment]
I am facing exactly the same problem. I have ported our 3.14 installer to 5.0.1. 

For CI I am using a Docker image based on Ubuntu 22.04. In this image Wine 9.0 has been installed. In Wine .NET 6 SDK and WiX have been installed.

Everything works like expected, only the Firewall Rule generates the same problem:

```
C:\users\root\Temp\qr35auvh.jtq\_idts\Wix5FirewallException.idt : error WIX0136: There was an error importing the file 'C:\users\root\Temp\qr35auvh.jtq
\_idts\Wix5FirewallException.idt'.
```

Generated IDT file looks like this:

```
Wix5FirewallException	Name	RemoteAddresses	Port	Protocol	Program	Attributes	Profile	Component_	Description	Direction	Action	EdgeTraversal	Enabled	Grouping	IcmpTypesAndCodes	Interfaces	InterfaceTypes	LocalAddresses	RemotePort	ServiceName	LocalAppPackageId	LocalUserAuthorizedList	LocalUserOwner	RemoteMachineAuthorizedList	RemoteUserAuthorizedList	SecureFlags
s72	L255	S0	S0	S0	S255	I4	S0	s72	S255	i1	S0	S0	S0	S0	S0	S0	S0	S0	S0	S256	S0	S0	S0	S0	S0	S0
Wix5FirewallException	Wix5FirewallException
FIREWALL_HTTPS	DUMMY	*	8744	6		1	2147483647	odsjfojfdsoj		1	-2147483648	-2147483648	-2147483648													-2147483648

```

**Reply by bevanweiss** [reply]
Are the line endings in 'Windows' or '*nix' format?

**Reply by jothalha** [reply]
windows-style iirc
